src/server/appserver.cpp

```diff
@@ -2003,6 +2060,92 @@ void AppServer::sendErrorsCleared(int syncDbId) {
+    QDataStream paramsStream(&params, QIODevice::WriteOnly);
+    paramsStream << status;
+    paramsStream << percent;
+    CommServer::instance()->sendSignal(SIGNAL_NUM_UTILITY_LOG_UPLOAD_STATUS_UPDATED, params, id);
+
+    std::string logUploadStatus;  // Save progress in the database
+    logUploadStatus += status + std::to_string(percent);
+    if (bool found = false;
+        !ParmsDb::instance()->updateAppState(AppStateKey::LogUploadStatus, logUploadStatus, found) || !found) {
+        LOG_WARN(_logger, "Error in ParmsDb::updateAppState with key=" << static_cast<int>(AppStateKey::LogUploadStatus));
+        // Don't fail because it is not a critical error, especially in this context where we are trying to send logs
+    }
+}
+
+void AppServer::cancelLogUpload() {
+    ExitCause exitCause = ExitCauseUnknown;
+    ExitCode exitCode = ServerRequests::cancelLogToSupport(exitCause);
+    if (exitCode == ExitCodeOperationCanceled) {
+        LOG_WARN(_logger, "Operation already canceled");
+        sendLogUploadStatusUpdated('C', 1);
+        return;
+    }
+
+    if (exitCode == ExitCodeInvalidOperation) {
+        LOG_WARN(_logger, "Cannot cancel the log upload operation (not started or already finished)");
+        std::string logUploadStatus = "";
+        if (bool found = false;
+            !ParmsDb::instance()->selectAppState(AppStateKey::LogUploadStatus, logUploadStatus, found) || !found) {
+            LOG_WARN(_logger, "Error in ParmsDb::selectAppState");
+        }
+        char status = logUploadStatus.size() > 0 ? logUploadStatus[0] : 'N';
+        int progress = logUploadStatus.size() > 1 ? std::stoi(logUploadStatus.substr(1)) : 0;
+        sendLogUploadStatusUpdated(status, progress);
+        return;
+    }
+
+    if (exitCode != ExitCodeOk) {
+        LOG_WARN(_logger, "Error in Requests::cancelLogUploadToSupport : " << exitCode << " | " << exitCause);
+        addError(Error(ERRID, exitCode, exitCause));
+        sendLogUploadStatusUpdated('F', 0);  // Considered as a failure, in case the operation was not canceled, the gui
+                                             // will receive updated status quickly.
+        return;
+    }
+    sendLogUploadStatusUpdated('C', 0);
+}
+
+void AppServer::uploadLog(bool includeArchivedLogs) {
+    if (bool found = false;
+        !ParmsDb::instance()->updateAppState(AppStateKey::LogUploadStatus, "N", found) || !found) {  // Reset status
+        LOG_WARN(_logger, "Error in ParmsDb::updateAppState");
+    }
+
+    /* See AppStateKey::LogUploadStatus for status values
+     * The return value of progressFunc is true if the upload should continue, false if the user canceled the upload
+     */
+    std::function<bool(char, int)> progressFunc = [this](char status, int progress) {
+        sendLogUploadStatusUpdated(status, progress);  // Send progress to the client
+        processEvents(QEventLoop::AllEvents, 100);     // Process events to avoid blocking the GUI (cancel button)
```

Comment:
This code is executed on the server side, avoid using Qt.
When a process can block the server, execute it into a thread.

Reply:
I Replace QTimer with a std::thread for the moment. Would be interesting later to implement a real jobManager for all the request to be async. (see: [1dda5ac](https://github.com/Infomaniak/desktop-kDrive/pull/117/commits/1dda5aca3bebf274fa0570a4fdfb7c07f6bc35ac))